### PR TITLE
TypedObjectTypingResult - use regular Map in place of ListMap, fields order no longer matters

### PR DESCRIPTION
**Pull request by rafa-minimal**
Let's have a consistent representation of TypedObject with keys sorted.

### CODE REVIEW

**Review by bartektartanus — APPROVED** [review]

**Review by arkadius — CHANGES_REQUESTED** [review]
Why? We have defined order, fields are in ListMap - taka a look at warning above the class

**Comment by rafa-minimal** [comment]
I didn't really understand the reasons behind it. Ok, let's make things simple. Merge will return a type definition with keys sorted. Seems the best option for me. WDYT?

**Review by arkadius — APPROVED** [review]
I think, that it LGTM :)

**Review by arkadius — CHANGES_REQUESTED** [review]
I have to revert my  :+1: I see that you changed how the result is typed, but you didn't change the implementation. If you change the implementation, the merge will change complexity from O(n+m) into O((n+m)log(n+m)). Do you think that it is worth it :thinking: (I'm not trying to state an opinion about that)

**Comment by rafa-minimal** [comment]
All right, it required a little bit of analysis, but we've come to a solution.

The reason `TypedObjectTypingResult` was using `ListMap` is that the frontend form was generated based on the `TypingResult` and we wanted to keep fields in order (see #1532). This is no longer the case (see `AvroSchemaBasedParameter`), so we can put back a regular `Map` in place of `ListMap`.

This change reverts #1532.

Additionally let's sort `TypedObjectTypingResult` keys on display:
- to keep it consistent,
- to make it easier to compare two types or find a particular field in a large type definition.

File: engine/flink/tests/src/test/resources/extractedTypes/defaultModel.json [hunk 1]
```diff
@@ -13128,7 +13128,7 @@
       ],
       "merge": [
         {
-          "description": "Merges two maps. Values in the first map will be overwritten with values from the second map if keys are the same",
+          "description": "Merges two maps. Values in the first map will be overwritten with values from the second map if keys are the same. The keys for the resulting type definition will be sorted.",
```
Comment (arkadius) [inline engine/flink/tests/src/test/resources/extractedTypes/defaultModel.json:13131]: let's remove this pat of text about sorted keys

**Review by arkadius — APPROVED** [review]
LGTM

File: engine/flink/tests/src/test/resources/extractedTypes/defaultModel.json (hunk 1 above)
Comment (rafa-minimal) [reply engine/flink/tests/src/test/resources/extractedTypes/defaultModel.json]: Thanks!

**Pull request merged**